superdesk/io/feeding_services/gmail.py

```diff
@@ -99,6 +103,32 @@ def authenticate(self, provider: dict, config: dict) -> imaplib.IMAP4_SSL:
         imap.authenticate("XOAUTH2", lambda __: auth_string.encode())
         return imap
 
+    def parse_extra(self, imap: imaplib.IMAP4_SSL, num: str, parsed_items: List[dict]) -> None:
+        """Add GMail labels to parsed_items"""
+        try:
+            # we use GMail IMAP Extensions
+            # https://developers.google.com/gmail/imap/imap-extensions#access_to_gmail_labels_x-gm-labels
+            _, data = imap.fetch(num, "(X-GM-LABELS)")
+            # it seems that there is nothing to help parsing in standard lib
+            # thus we use some regex to get our labels
+            data_bytes = data[0]
+            if not isinstance(data_bytes, bytes):
+                raise ValueError(f"Unexpected data type: {type(data_bytes)}")
+            data_str = data_bytes.decode("utf-7")
+            match_labels_str = RE_LABELS_STR.search(data_str)
+            if match_labels_str is None:
+                raise ValueError(f"Can't find the expected label string in data: {data_str:r}")
+            labels_str = match_labels_str.group(1)
+            labels = [
+                (m.group("quoted") or m.group("unquoted")).replace('\\"', '"') for m in RE_LABEL.finditer(labels_str)
+            ]
+            for parsed_item in parsed_items:
+                subjects = parsed_item.setdefault("subject", [])
+                for label in labels:
+                    subjects.append({"name": label, "qcode": label, "scheme": "label"})
```

Comment:
`label` seems kind of generic scheme and potentially conflicting, I would call it `gmail_label`

Reply:
I actually did it on purpose, I thought a generic label would be a good thing if similar labels are used with several ingest formats. But it's not a problem to change if you think it's better, I'll update it.